src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -657,6 +657,15 @@ void GcsActorManager::ReconstructActor(const ActorID &actor_id, bool need_resche
         }));
     gcs_actor_scheduler_->Schedule(actor);
   } else {
+    // For detached actors, make sure to clean up their names.
```

Comment:
I think a better place for this cleanup code is in `DestroyActor`. Actually, it seems like maybe this code should call `DestroyActor` if the actor was detached.

Can you also add a unit test for the GcsActorManager, similar to the Python tests?

Reply:
When I read code, it seems like `DestroyActor` was called only for children. https://github.com/ray-project/ray/blob/c773824f4f22a14375a2451264d1b5708addd0f9/src/ray/gcs/gcs_server/gcs_actor_manager.cc#L563. Would you happen to know why? 

Also, I will add an unit test here.

Reply:
Actually, one other easy solution is to remove the named entry when detached actor is created (same mechanism as non-gcs-actor-mangement-enabled).

Reply:
Oh yes, right now `DestroyActor` is only called for non-detached actors. There are two cases for cleaning up a non-detached actor. Either:
1. The actor's owner dies (this is the case that you saw).
2. If the actor's owner says that the handle has gone out of scope.

We should also call it for detached actors, otherwise we'll leak memory. I think it makes sense to call it here, once we know for sure that the actor is dead.

Reply:
I'm not sure what you mean about removing the named entry. Do you mean we should remove an old entry if it has a duplicate name? I'm not sure if that will work since we'd have to figure out if the old entry is still valid.

Reply:
Ah got it. Then it is like detached actors weren't destroyed properly so far. I will add DestroyActor here.

Also, what I meant by named entry is this;

Before creating a new detached actor, we check if the name exists through calling HandleGetByName or something. I can add a check inside that handler that basically checks if the named actor state is DEAD or not. If the state is DEAD, I treat it as the name is available. Does it make sense to you? (I might miss something)

Reply:
Ah I see, that does make sense now. I think it probably is not necessary, though, once we add the call to DestroyActor here (then it should never be true that the name exists && the named actor's state is DEAD).

Reply:
Yeah I agree that's a clearer approach. Let me try adding destroy actor here and see if it breaks anything haha.